protocol/contracts/lpp/src/lpp.rs

```diff
@@ -891,6 +891,135 @@ mod test {
         assert_eq!(withdraw, Coin::new(1110));
     }
 
+    #[test]
+    fn open_and_repay_with_overflow() {
+        let lpp_balance: Amount = 10_000_000;
+        let mut deps = testing::mock_dependencies_with_balance(&[coin_cw(lpp_balance)]);
+        let mut env = testing::mock_env();
+        let admin = Addr::unchecked("admin");
+        let lease_addr_1 = Addr::unchecked("loan1");
+        let lease_addr_2 = Addr::unchecked("loan2");
+        env.block.time = Timestamp::from_nanos(0);
+        let lease_code_id = Code::unchecked(123);
+
+        grant_admin_access(deps.as_mut(), &admin);
+
+        Config::new_unchecked(
+            lease_code_id,
+            InterestRate::new(
+                BASE_INTEREST_RATE,
+                UTILIZATION_OPTIMAL,
+                ADDON_OPTIMAL_INTEREST_RATE,
+            )
+            .expect("Couldn't construct interest rate value!"),
+            DEFAULT_MIN_UTILIZATION,
+        )
+        .store(deps.as_mut().storage)
+        .expect("Failed to store Config!");
+        Total::<TheCurrency>::new()
+            .store(deps.as_mut().storage)
+            .expect("can't initialize Total");
+
+        let mut lpp = LiquidityPool::<TheCurrency>::load(deps.as_mut().storage)
+            .expect("can't load LiquidityPool");
+
+        let borrow_loan1 = Coin::<TheCurrency>::new(5_458_329);
+        let loan1_annual_interest_rate = Percent::from_permille(137);
+        let mut loan1 = Loan {
+            principal_due: borrow_loan1,
+            annual_interest_rate: loan1_annual_interest_rate,
+            interest_paid: env.block.time,
+        };
+
+        Loan::open(&mut deps.storage, lease_addr_1.clone(), &loan1).unwrap();
+
+        lpp.total
```

Comment:
what interface do we test against? why do we break the encapsulation reaching to `lpp.total`?